File: gems/decomplex/examples/swift/implicit-control-flow.swift
class FlowExample {
  var status = 0
  var valid = false
  var done = false
  func prepare() { self.status = 1 }
  func validate() { self.valid = self.status == 1 }
  func commit() { self.done = self.valid }
  func ok1() { self.prepare(); self.validate(); self.commit() }
  func ok2() { self.prepare(); self.validate(); self.commit() }
  func ok3() { self.prepare(); self.validate(); self.commit() }
  func ok4() { self.prepare(); self.validate(); self.commit() }
  func drift() { self.validate(); self.prepare(); self.commit() }
}

File: gems/decomplex/examples/ruby/implicit-control-flow.rb
# frozen_string_literal: true

class FlowExample
  def prepare; self.status = :ready; end
  def validate; @valid = status == :ready; end
  def commit; self.done = @valid; end

  def ok1; prepare; validate; commit; end
  def ok2; prepare; validate; commit; end
  def ok3; prepare; validate; commit; end
  def ok4; prepare; validate; commit; end
  def drift; validate; prepare; commit; end
end
